uportal-war/src/main/java/org/jasig/portal/jgroups/protocols/DAO_PING.java

```diff
@@ -101,17 +108,18 @@ public void destroy() {
             log.info("No PingDao set, returning empty set for current cluster members");
             return Collections.emptyList();
         }
-        
-        //Get the current cluster members
-        final Map<Address, PhysicalAddress> existing_mbrs = pingDao.getAddresses(clusterName);
 
         //Add our address to the store
         getAndSavePhysicalAddress(clusterName);
-        
+
+        //Get the current cluster members
+        final Map<Address, PhysicalAddress> existing_mbrs = pingDao.getAddresses(clusterName);
+
```

Comment:
The DAO implementation retrieves all addresses from the table, including the local address if it is present. Also, the method getAndSavePhysicalAddress() is called on a timer. The original order may play a part on the very first call to fetchClustermembers(), which should then be documented.